Source/WebKit/Shared/Cocoa/RevealItem.h

```diff
@@ -27,24 +27,44 @@
 
 #if ENABLE(REVEAL)
 
-#import "ArgumentCoders.h"
-
 #import <wtf/RetainPtr.h>
 
 OBJC_CLASS RVItem;
 
+typedef struct _NSRange NSRange;
+typedef unsigned long NSUInteger;
+
 namespace WebKit {
 
+struct RevealItemRange {
+    RevealItemRange() = default;
+    RevealItemRange(NSRange);
+    RevealItemRange(NSUInteger loc, NSUInteger len)
+        : location(loc)
+        , length(len)
+    {
+    }
+
+    NSUInteger location { 0 };
+    NSUInteger length { 0 };
```

Comment:
Is there a reason you're choosing to store this as the parts of an NSRange instead of an NSRange?

Reply:
Yes! Teaching the serializer generator to handle a raw C struct typedef like NSRange turned out to be extremely painful!

If that capability gets added to the generator in the future, we can ditch the range wrapper and just go to a raw NSRange

Reply:
Maybe add a note to that effect? And/or file a radar?